plasmapy/analysis/swept_langmuir/ion_saturation_current.py

```diff
@@ -0,0 +1,228 @@
+"""
+Functionality for determining the ion-saturation current of a Langmuir sweep.
+"""
+__all__ = ["find_ion_saturation_current", "ISatExtras"]
+__aliases__ = ["find_isat_"]
+
+import numbers
+import numpy as np
+
+from typing import Tuple, NamedTuple, Union
+
+from plasmapy.analysis import fit_functions as ffuncs
+from plasmapy.analysis.swept_langmuir.helpers import check_sweep
+
+__all__ += __aliases__
+
+
+class ISatExtras(NamedTuple):
+    """
+    Create a tuple containing the extra parameters calculated by
+    `find_ion_saturation_current`.
+    """
+    rsq: Union[float, None]
+    """
+    Alias for field number 0, the r-squared value of the ion-saturation
+    curve fit.
+    """
+
+    fitted_func:  Union[ffuncs.AbstractFitFunction, None]
+    """
+    Alias for field number 1, the :term:`fit-function` fitted during
+    the ion-saturation curve fit.
+    """
+
+    fitted_indices: Union[slice, None]
+    """
+    Alias for field number 2, the indices used in the ion-saturation
+    curve fit.
+    """
+
+
+def find_ion_saturation_current(
+    voltage: np.ndarray,
+    current: np.ndarray,
+    *,
+    fit_type: str = "exp_plus_linear",
+    current_bound: float = None,
+    voltage_bound: float = None,
+) -> Tuple[ffuncs.Linear, ISatExtras]:
+    """
+    Determines the ion-saturation current (:math:`I_{sat}`) for a given
+    current-voltage (IV) curve obtained from a swept Langmuir probe.
+    The current collected by a Langmuir probe reaches ion-saturation
+    when the probe is sufficiently biased so the influx of electrons is
+    completely repelled leading to only the collection of ions.  (For
+    additional details see the **Notes** section below.)
+
+    **Aliases:** `find_isat_`
+
+    Parameters
+    ----------
+
+    voltage: `numpy.ndarray`
+        1-D numpy array of monotonically increasing probe biases
+        (should be in volts).
+
+    current: `numpy.ndarray`
+        1-D numpy array of probe current (should be in amperes)
+        corresponding to the ``voltage`` array.
+
+    fit_type: `str`
+        The type of curve (:term:`fit-function`) to be fitted to the
+        Langmuir trace, valid options are listed below.
+        (DEFAULT ``"exp_plus_linear"``)
```

Comment:
```suggestion
        Defaults to ``"exp_plus_linear"``.
```
One thing I learned somewhat recently is that using [all caps](https://en.wikipedia.org/wiki/All_caps) has generally been found to reduce legibility.

Reply:
While I generally agree with the statement about the readability of all caps, I don't think it applies here.  The all caps statement applies to general writings (emails, books, journals, etc.), but we have something a little more specialized here.  One of my biggest issues with documenting functionality is that default values are buried within the documentation, making it rather difficult to find.  This is my issue with the suggested replacement.  Now, while I'm not particularly thrilled with my own implementation, I do think it provides better transparency.

My current take is to leave it how it is, and change it when we finally decided on a standard to implement throughout the package.